server/src/server.rs

```diff
@@ -329,22 +465,85 @@ impl Builder<Identity, Identity> {
 	}
 }
 
+impl<RpcMiddleware: Clone, HttpMiddleware: Clone> TowerServiceBuilder<RpcMiddleware, HttpMiddleware> {
+	/// Build a tower service.
+	pub fn build(
+		&self,
+		methods: impl Into<Methods>,
+		stop_handle: StopHandle,
+	) -> TowerService<RpcMiddleware, HttpMiddleware> {
+		let conn_id = self.conn_id.fetch_add(1, std::sync::atomic::Ordering::Relaxed);
```

Comment:
this could have been a u32 but it's so annoying with mutable stuff in  FnMut closure's such as `hyper::service_fn`